lib/iris/experimental/ugrid.py

```diff
@@ -2680,6 +2681,257 @@ def face_node(self):
+            raise ValueError(msg)
+
+    @property
+    def climatological(self):
+        """The 'climatological' of a MeshCoord is always 'False'."""
+        return False
+
+    @climatological.setter
+    def climatological(self, value):
+        if value:
+            msg = "Cannot set 'climatological' on a MeshCoord."
+            raise ValueError(msg)
+
+    def __getitem__(self, keys):
+        # Disallow any sub-indexing, permitting *only* "self[:,]".
+        # We *don't* intend here to support indexing as such : the exception is
+        # just sufficient to enable cube slicing, when it does not affect the
+        # mesh dimension.  This works because Cube.__getitem__ passes us keys
+        # "normalised" with iris.util._build_full_slice_given_keys.
+        if keys != (slice(None),):
+            msg = "Cannot index a MeshCoord."
+            raise ValueError(msg)
+
+        # Translate "self[:,]" as "self.copy()".
+        return self.copy()
+
+    def copy(self, points=None, bounds=None):
+        """
+        Make a copy of the MeshCoord.
+
+        Kwargs:
+
+        * points, bounds (array):
+            Provided solely for signature compatibility with other types of
+            :class:`~iris.coords.Coord`.
+            In this case, if either is not 'None', an error is raised.
+
+        """
+        # Override Coord.copy, so that we can ensure it does not duplicate the
+        # Mesh object (via deepcopy).
+        # This avoids copying Meshes.  It is also required to allow a copied
+        # MeshCoord to be == the original, since for now Mesh == is only true
+        # for the same identical object.
+
+        # FOR NOW: also disallow changing points/bounds at all.
+        if points is not None or bounds is not None:
+            msg = "Cannot change the content of a MeshCoord."
+            raise ValueError(msg)
+
+        # Make a new MeshCoord with the same args :  The Mesh is the *same*
+        # as the original (not a copy).
+        new_coord = MeshCoord(
+            mesh=self.mesh, location=self.location, axis=self.axis
+        )
+        return new_coord
+
+    def __deepcopy__(self, memo):
+        """
+        Make this equivalent to "shallow" copy, returning a new MeshCoord based
+        on the same Mesh.
+
+        Required to prevent cube copying from copying the Mesh, which would
+        prevent "cube.copy() == cube" :  see notes for :meth:`copy`.
+
+        """
+        return self.copy()
+
+    # Override _DimensionalMetadata.__eq__, to add 'mesh' comparison into the
+    # default implementation (which compares metadata, points and bounds).
+    # This is needed because 'mesh' is not included in our metadata.
+    def __eq__(self, other):
+        eq = super().__eq__(other)
+        # warning may return NotImplemented or a numpy boolean.
+        if eq is not NotImplemented and eq:
+            # Ensures that 'other' is also a MeshCoord
+            eq = self.mesh == other.mesh
```

Comment:
@pp-mo You've not addressed this yet

Reply:
I hadn't spotted this.
I think we need to do more : "Ideally", if we accept that the meshes are equivalent, then we know that the points, bounds and basic coord properties (names, units, attributes) *must* all be equal
Except, actually, we **_don't_**, since the Mesh might have changed between creating the two MeshCoords :cry: 

So, I want to say we should *never* check anything but the Mesh/location/axis, certainly not points, bounds, attributes etc...
But owing to the limitations on the 'dynamic'-ness, this isn't actually quite solid.

So, what do you think we **_should_** we do with this?

Reply:
Stop press: 
I'm now proposing we compare mesh/location/axis.
That makes logical sense, though it is not actually quite correct in view of the possibility for the Mesh to change.